From 9fbeb8fddae98ee80d9195a78cef3ca3d510c927 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Tue, 3 Oct 2017 13:22:56 +0200
Subject: [PATCH] fixxref/rebase: Don't depend on glib being installed

As is the case when you build docs for glib itself.
---
 gtkdoc/fixxref.py | 28 ++++++++++++++++------------
 gtkdoc/rebase.py  | 11 ++++++++---
 2 files changed, 24 insertions(+), 15 deletions(-)

diff --git a/gtkdoc/fixxref.py b/gtkdoc/fixxref.py
index a7029cb..7b1901f 100755
--- a/gtkdoc/fixxref.py
+++ b/gtkdoc/fixxref.py
@@ -71,18 +71,22 @@ def Run(options):
 
     # We scan the directory containing GLib and any directories in GNOME2_PATH
     # first, but these will be overriden by any later scans.
-    dir = common.GetModuleDocDir('glib-2.0')
-    if os.path.exists(dir):
-        # Some predefined link targets to get links into type hierarchies as these
-        # have no targets. These are always absolute for now.
-        Links['GBoxed'] = dir + '/gobject/gobject-Boxed-Types.html'
-        Links['GEnum'] = dir + '/gobject/gobject-Enumeration-and-Flag-Types.html'
-        Links['GFlags'] = dir + '/gobject/gobject-Enumeration-and-Flag-Types.html'
-        Links['GInterface'] = dir + '/gobject/GTypeModule.html'
-
-        if dir != options.html_dir:
-            logging.info('Scanning GLib directory: %s', dir)
-            ScanIndices(dir, (re.search(prefix_match, dir) is None))
+    try:
+        dir = common.GetModuleDocDir('glib-2.0')
+    except subprocess.CalledProcessError:
+        pass
+    else:
+        if os.path.exists(dir):
+            # Some predefined link targets to get links into type hierarchies as these
+            # have no targets. These are always absolute for now.
+            Links['GBoxed'] = dir + '/gobject/gobject-Boxed-Types.html'
+            Links['GEnum'] = dir + '/gobject/gobject-Enumeration-and-Flag-Types.html'
+            Links['GFlags'] = dir + '/gobject/gobject-Enumeration-and-Flag-Types.html'
+            Links['GInterface'] = dir + '/gobject/GTypeModule.html'
+
+            if dir != options.html_dir:
+                logging.info('Scanning GLib directory: %s', dir)
+                ScanIndices(dir, (re.search(prefix_match, dir) is None))
 
     path = os.environ.get('GNOME2_PATH')
     if path:
diff --git a/gtkdoc/rebase.py b/gtkdoc/rebase.py
index d6affe3..44ca5a9 100755
--- a/gtkdoc/rebase.py
+++ b/gtkdoc/rebase.py
@@ -30,6 +30,7 @@ from six import iteritems, iterkeys
 import logging
 import os
 import re
+import subprocess
 
 from . import common
 
@@ -63,9 +64,13 @@ def run(options):
                 log(options, "Prepending GNOME2_PATH directory:", dir)
                 other_dirs = [dir] + other_dirs
 
-    dir = common.GetModuleDocDir('glib-2.0')
-    log(options, "Prepending GLib directory", dir)
-    other_dirs = [dir] + other_dirs
+    try:
+        dir = common.GetModuleDocDir('glib-2.0')
+    except subprocess.CalledProcessError:
+        pass
+    else:
+        log(options, "Prepending GLib directory", dir)
+        other_dirs = [dir] + other_dirs
 
     # Check all other dirs, but skip already scanned dirs ord subdirs of those
 
-- 
2.14.2

